### DISCUSSION TITLE
How to access storage files with third-party applications (e.g., AWS Transcribe)

### DISCUSSION THREAD

**Discussion by davidaventimiglia-professional** [supabase/supabase#7744] [Questions] [answered]
Hello,

How can we access Storage files with third-party apps, such as [AWS Transcribe](https://docs.aws.amazon.com/transcribe/latest/dg/what-is.html)?  The reason I ask is this.

Suppose I set up Supabase Storage and start uploading audio files.  It's my understanding that (currently) Supabase uses AWS S3 to store those files.  Meanwhile, AWS Transcribe (which translates audio to text) jobs take input from AWS S3.  It sure would be useful if I could set up transcription jobs to transcribe files that are uploaded to Supabase Storage.  Unfortunately, currently I have no idea how to do that.  More generally, I have no idea how to _operate_ on files added to Supbase Storage.  The [Supabase Docs](https://supabase.com/docs/guides/storage) only seem to address the accessing of Storage files from SQL, within a Supabase database.  How do I access those files from third-party apps, in general?  Thanks so much.

Best,
David

**Comment by GaryAustin1** [comment, marked as answer]
Currently the only way to access storage files is with storage download() or with urls.  
The current system uses a shared s3 bucket behind the scenes for multiple instances, so there is no general access using any sort of s3 interface.
I've seen discussions about future bring your own s3 compatible storage, but no timelines.

**Reply by davidaventimiglia-professional** [reply]
I suppose I could use storage download() to get a copy of the file, copy it over to _another_ S3 bucket, have AWS do the transcription, and then process the results.  It's a bit redundant with the S3 storage, and it involves writing code and some extra steps but it's not the end of the world. Thanks.
